tests/unit_tests/weightwindows/test.py

```diff
@@ -0,0 +1,169 @@
+    mesh.upper_right = (240, 240, 240)
+    mesh.dimension = (5, 10, 15)
+
+    mesh_filter = openmc.MeshFilter(mesh)
+
+    e_bnds = [0.0, 0.5, 2E7]
+    energy_filter = openmc.EnergyFilter(e_bnds)
+
+    particle_filter = openmc.ParticleFilter(['neutron', 'photon'])
+
+    tally = openmc.Tally()
+    tally.filters = [mesh_filter, energy_filter, particle_filter]
+    tally.scores = ['flux']
+
+    model.tallies.append(tally)
+
+    return model
+
+
+def test_weightwindows(model):
+
+    ww_files = ('ww_n.txt', 'ww_p.txt')
+    cwd = Path(__file__).parent.absolute()
+    filepaths = [cwd / Path(f) for f in ww_files]
+
+    with cdtemp(filepaths):
+        # run once with variance reduction off
+        model.settings.weight_windows_on = False
+        analog_sp = model.run()
+        os.rename(analog_sp, 'statepoint.analog.h5')
+
+        # weight windows
+
+        # load pre-generated weight windows
+        # (created using the same tally as above)
+        ww_n_lower_bnds = np.loadtxt('ww_n.txt')
+        ww_p_lower_bnds = np.loadtxt('ww_p.txt')
+
+        # create a mesh matching the one used
+        # to generate the weight windows
+        ww_mesh = openmc.RegularMesh()
+        ww_mesh.lower_left = (-240, -240, -240)
+        ww_mesh.upper_right = (240, 240, 240)
+        ww_mesh.dimension = (5, 6, 7)
+
+        # energy bounds matching those of the
+        # generated weight windows
+        e_bnds = [0.0, 0.5, 2E7]
+
+        ww_n = openmc.WeightWindows(ww_mesh,
+                                    ww_n_lower_bnds,
+                                    None,
+                                    10.0,
+                                    e_bnds)
+
+        ww_p = openmc.WeightWindows(ww_mesh,
+                                    ww_p_lower_bnds,
+                                    None,
+                                    10.0,
+                                    e_bnds)
+
+        model.settings.weight_windows = [ww_n, ww_p]
+
+
+        # run again with variance reduction on
+        model.settings.weight_windows_on = True
+        ww_sp = model.run()
+        os.rename(ww_sp, 'statepoint.ww.h5')
+
+        # load both statepoints and examine results
+        asp = openmc.StatePoint('statepoint.analog.h5')
+        wsp = openmc.StatePoint('statepoint.ww.h5')
+
+        analog_tally = asp.tallies[1]
+        ww_tally = wsp.tallies[1]
+
+        def compare_results(particle, analog_tally, ww_tally):
+
+            # get values from each of the tallies
+            an_mean = analog_tally.get_values(filters=[openmc.ParticleFilter],
+                                              filter_bins=[(particle,)])
+            ww_mean = ww_tally.get_values(filters=[openmc.ParticleFilter],
+                                          filter_bins=[(particle,)])
+
+            # expect that more bins were scored with weight windows than
+            # the analog run
+            assert np.count_nonzero(an_mean) < np.count_nonzero(ww_mean)
```

Comment:
Should we also compare the mean results of a particular bin to assure the analog and ww runs produce the same results? You can do this by increasing the mesh tally resolution to something like (15, 15, 15) rather than (5, 10, 15) and compare the bin at (7, 7, 7) which should contain the source.

Reply:
Good call, @eepeterson. While the flux in that region is likely to be dominated by the source particles, which will be the same with our without weight windows applied, it's still a good check to have. Following your suggestion, I went ahead and upped the dimensions of the mesh to (3, 5,  7). The lower dimensions help to ensure we get enough samples in the source bin to make our statistical check valid since we're only running 1000 particles in this test. Right now I'm checking that the analog and weight window runs are with in 2 combined standard deviations of each other. Let me know If you have additional metrics you'd like to see there.

Reply:
This is great, thanks for the update. I realize it would probably take a prohibitively long amount of time to sample a region that actually depends on the weight windows substantially, but this is good enough for now.